Tests/YSwiftTests/YSubscriptionTests.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+
+import Combine
+import XCTest
+@testable import YSwift
+
+class YSubscriptionTests: XCTestCase {
+    var document: YDocument!
+    var text: YText!
+    
+    override func setUp() {
+        document = YDocument()
+        text = document.getOrCreateText(named: "test")
+    }
+    
+    override func tearDown() {
+        document = nil
+        text = nil
+    }
+    
+    private func test_cancel(cancel: (inout YSubscription?) -> Void) {
+        // Create an object (it can be of any type), and hold both
+        // a strong and a weak reference to it
+        var object = NSObject()
+        weak var weakObject = object
+
+        var subscription: YSubscription? = text.observe { _ in
+            // Capture the object in the closure (note that we need to use
+            // a capture list like [object] above in order for the object
+            // to be captured by reference instead of by pointer value)
+            _ = object
+        }
+
+        // Explicit unobserving, to prevent leaking
+        cancel(&subscription)
+
+        // When we re-assign our local strong reference to a new object the
+        // weak reference should become nil, since the closure should
+        // have been run and removed at this point
+        // Because we did explicitly unobserve/unsubscribe at this point.
+        object = NSObject()
+        XCTAssertNil(weakObject)
+    }
+    
+    func test_cancel_deinit() {
+        test_cancel { subscription in
+            subscription = nil
+        }
+    }
+    
+    func test_cancel_explicit() {
+        test_cancel { subscription in
+            subscription?.cancel()
+        }
+    }
+    
+    // This causes memory to keep growing and I am not sure why
+    // I checked that UniFFI YSubscription struct is being dropped accordingly
+    // I also verified that the memory increase also happened when we used subscriptionIds and unobserve
```

Comment:
As I describe here, I am not sure why setting up observations increase memory, even before this change. Any thoughts on this?

Reply:
In new model subscriptions themselves keep strong reference to a callback, while internal callback manager keeps a list of weak references. These references are not released right away, but in one of two cases:

1. When we drop an iterator used for triggering the callbacks.
2. When we add a new callback: since it's happening when using atomic update, we need to prepare a copy of older callbacks, so we also get rid of the weak refs without the strong ones.

You can add a simple text.insert operation to confirm if callback trigger is performing the cleanup. You could also try to cancel subscription explicitly. If none of these works, then probably we have memory leak problem in core lib.

Reply:
I tried this in a test and it doesn't prevent memory going up.
```swift
    func test_allocateTooMany() {
        for i in 1...1000000000 {
            let s = text.observe { _ in }
            text.unobserve(s)
            
            if i % 1000000 == 0 {
                print("\(Date()) Iterations: \(i) Resident Size: \(residentSize() ?? 0)")
                text.insert("a", at: 0)
            }
        }
    }
```
Not sure if the leak is in Yswift side or Yrs though.

Reply:
Note the above test was run before this PR. For this PR, the equivalent test is this one:
```swift
    func testallocateTooMany() {
        for i in 1...1000000000 {
            let s = text.observe { _ in }
            s.cancel()
            
            if i % 1000000 == 0 {
                print("\(Date()) Iterations: \(i) Resident Size: \(residentSize() ?? 0)")
                text.insert("a", at: 0)
            }
        }
    }
    
```
In both memory keeps going up.

Reply:
I suspect you've spotted an issue outside of what changed in this PR - if you're game, let's separate the two and hunt the memory issue we're seeing here separately from these changes.